### PR TITLE
scripts/bump-pebble.sh: use current branch, don't use pebble repo

### PR DESCRIPTION
**Pull request by RaduBerinde**
#### scripts/bump-pebble.sh: add links for commits

Make the commits into links so they can easily be inspected by
reviewers.

Epic: none
Release note: None

#### scripts/bump-pebble.sh: don't use local pebble repo

Instead of assuming there is a local pebble repository, we now create
a temporary directory and clone a repository there.  Pebble is small
enough so this does not take long.

Epic: none
Release note: None

#### scripts/bump-pebble.sh: use current branch

We simplify the script to use the current branch and only create a new
branch if we are on master or the release branch. It is now the user's
responsibility to make sure the branch is up to date before running.

Epic: none
Release note: None

### CODE REVIEW

File: scripts/bump-pebble.sh [hunk 1]
```diff
@@ -63,9 +62,25 @@ popd
 
 # Ensure the local Pebble release branch is up-to-date with upstream,
 # and grab the desired Pebble SHA.
+PEBBLE_DIR=$(mktemp -d)
+trap "rm -rf $PEBBLE_DIR" EXIT
+
+git clone --no-checkout "$PEBBLE_UPSTREAM_URL" "$PEBBLE_DIR"
```
Comment (dt) [inline scripts/bump-pebble.sh:41]: should we make this shallow / depth = 50 too?

File: scripts/bump-pebble.sh [hunk 2]
```diff
@@ -1,107 +1,95 @@
...
     echo "Error: $NEW_SHA is not an ancestor of the pebble branch $PEBBLE_BRANCH" >&2
     exit 1
   fi
-  echo "Using provided sha $NEW_SHA."
+  echo "Using provided SHA $NEW_SHA."
+fi
+
+# Sanity check: the old SHA should be an ancestor of the new SHA.
+if ! git merge-base --is-ancestor $OLD_SHA $NEW_SHA; then
+  echo "Error: current pebble SHA $OLD_SHA is not an ancestor of $NEW_SHA. Is the current branch based on $BRANCH?" >&2
```
Comment (dt) [inline scripts/bump-pebble.sh:61]: should this be PEBBLE_BRANCH?

**Review by dt — APPROVED** [review]

**Review by RaduBerinde — COMMENTED** [review]
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/116340#-:-NlZOp-qD1elYVIxsfHE:b-814g8k)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @dt and @nicktrav)

___
*[`scripts/bump-pebble.sh` line 68 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/116340#gh-1425689019:-NlZNtro7fDOyz_XKO98:b70o7xc) ([raw file](https://github.com/cockroachdb/cockroach/blob/b28bcd9a0b714bb50eea8364a82e03a59595963b/scripts/bump-pebble.sh#L68)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

should we make this shallow / depth = 50 too?
</blockquote></details>

I tried it (with `--no-single-branch` since we may need a release branch) and it's not visibly faster, I'll keep it the same so we don't have to worry about corner cases.
___
*[`scripts/bump-pebble.sh` line 61 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/116340#gh-1425692741:-NlZMHvIBCYd6qVhpC-a:biq0taw) ([raw file](https://github.com/cockroachdb/cockroach/blob/2dcb1a26c75cd81d9458631634ab82d55ba2380f/scripts/bump-pebble.sh#L61)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

should this be PEBBLE_BRANCH? 
</blockquote></details>

It's referring to the current branch in the cockroach repo..  But I don't know how this could happen since whatever branch we are on should contain its own version of this script with the correct corresponding $PEBBLE_BRANCH. I simplified the message.

**Pull request merged**